src/functionaltest/java/com/ericsson/ei/frontend/pageobjects/SubscriptionPage.java

```diff
@@ -0,0 +1,149 @@
+        WebElement saveBtn = wait.until(ExpectedConditions.elementToBeClickable(By.id("btnSave")));
+        saveBtn.click();
+    }
+
+    public void clickKVbtn(String loc) {
+        WebElement kvBtn = wait.until(ExpectedConditions.elementToBeClickable(By.id(loc)));
+        kvBtn.click();
+    }
+
+    public void selectDropdown(String loc, String value) {
+        WebElement selectEle = wait.until(ExpectedConditions.elementToBeClickable(By.id(loc)));
+        Select dropdown = new Select(selectEle);
+        dropdown.selectByVisibleText(value);
+    }
+
+    public String getValueFromSelect() {
+        WebElement selectNotificationType = wait
+                .until(ExpectedConditions.elementToBeClickable(By.id("notificationType")));
+        Select dropdown = new Select(selectNotificationType);
+        return dropdown.getFirstSelectedOption().getText();
+    }
+
+    public String getValueFromElement() {
+        WebElement metaTxt = wait.until(ExpectedConditions.elementToBeClickable(By.id("metaData")));
+        return metaTxt.getAttribute("value");
+    }
+
+    public void addFieldValue(String loc, String value) {
+        WebElement ele = wait.until(ExpectedConditions.elementToBeClickable(By.id(loc)));
+        ele.clear();
+        ele.sendKeys(value);
+    }
+
+    public void clickUploadSubscriptionFunctionality(String filePath, String subUploadResponse)
+            throws ClientProtocolException, IOException, InterruptedException {
+        CloseableHttpResponse responseData = this.createMockedHTTPResponse(subUploadResponse, 200);
+        Mockito.doReturn(responseData).when(mockedHttpClient).execute(
+                Mockito.argThat(request -> ((HttpRequestBase) request).getURI().toString().contains("subscriptions")));
+
+        WebElement uploadInputField = driver.findElement(By.id("upload_sub"));
+        uploadInputField.sendKeys(filePath);
+    }
+
+    public Boolean presenceOfClickGetTemplateButton() {
+        try {
+            driver.findElement(By.className("get_subscription_template"));
+            return true;
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+    }
+
+    public void clickDownloadGetTemplate(String responseData)
+            throws ClientProtocolException, IOException, InterruptedException {
+        CloseableHttpResponse response = this.createMockedHTTPResponse(responseData, 200);
+        Mockito.doReturn(response).when(mockedHttpClient).execute(Mockito.argThat(
+                request -> ((HttpRequestBase) request).getURI().toString().contains("/subscriptionsTemplate")));
+        WebElement getTemplateButton = driver.findElement(By.className("get_subscription_template"));
+        getTemplateButton.click();
+    }
+    
+    public void clickViewBtn() {
+        WebElement viewBtn = wait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[contains(text(),'View')]")));
+        viewBtn.click();
```

Comment:
Will this xpath lookup have impact if html tag order in code changes?
I thinks its better to lookup html tags/DOM objects by the Id tag.

Reply:
Well, it is independent of path... xpth is not absolute path so it need not be changed with change in order.
You may notice that mostly the "ids" are used for localization but xpath is used when it is convenient.

Reply:
Ok. I think we should rely on ids where it is possible, so test wont't break if some new html tags is added. But if this is not impacted by that, then I think this is a OK solution.